### DISCUSSION TITLE
Accesssing Open ID Connect clients protected by Authelia from local domain

### DISCUSSION THREAD

**Discussion by sarpuser** [authelia/authelia#7328] [Q&A] [answered]
If I set up Open ID Connect for something like portainer, can I still access it by using <hostname>.local:<portainerPort>? I am assuming this won't work since the reverse proxy would not be involved and also the requests would not be coming from mydomain.com.

**Comment by glime-ai** [comment, marked as answer]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Yes, you can still access Portainer using `<hostname>.local:<portainerPort>` directly, but this method bypasses the OpenID Connect authentication provided by Authelia when set up through a reverse proxy. Accessing Portainer directly in this way means you are not using the security layer that OpenID Connect provides, which is typically enforced through the reverse proxy configuration.

When you access Portainer directly by its hostname and port, you are essentially accessing the service at the network level without any of the authentication or authorization checks that would be performed by the reverse proxy and Authelia. This could be useful for local testing or internal network scenarios where security controls are handled differently.

However, for secure production environments, especially those exposed to the internet, it is recommended to access services through the reverse proxy to ensure that authentication and security policies are enforced. If your setup requires accessing services like Portainer directly using `.local` addresses, consider implementing necessary security measures at the service level or using network-level controls to restrict access.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).
